pyresample/geometry.py

```diff
@@ -1012,7 +1047,33 @@ def compute_domain(self, corners, resolution=None, shape=None):
                        corners[3] + y_resolution / 2)
         return area_extent, width, height
 
-    def freeze(self, lonslats=None, resolution=None, shape=None, proj_info=None):
+    def _update_corners_for_full_extent(self, corners, shape, resolution, projection):
+        corners = list(corners)
+        if corners[0] is not None:
+            return corners
+        aou = self._get_crs_area_of_use(projection)
+        if shape is not None:
+            width = shape[1]
+            x_resolution = (aou.east - aou.west) / width
+            corners[0] = aou.west + x_resolution / 2.0
+            corners[2] = aou.east - x_resolution / 2.0
+        else:
+            x_resolution = resolution[0]
+            corners[0] = aou.west + x_resolution / 2.0
+            corners[2] = aou.east - x_resolution / 2.0
```

Comment:
This looks quite similar to the `if shape` part in `compute_domain`. Could it be factorized, maybe by converting to CRS earlier and setting an area of use if there isn't already one?

Reply:
I thought about this for a long time when I originally wrote it and the only other option that I liked was to restructure `compute_domain` so it computes `extents` earlier (in which case area of use could be used directly), but I felt like it made the logic more complex and I think you end up with more code. I'm not seeing an easy way to do this.

Reply:
I spent some time trying to clean this usage up and I just can't find anything that I'm happy with. It can be restructured into multiple if/elif blocks and it cleans up the number of calculations to get the final extent, but makes the logic for everything else really confusing. You end up with duplicate code that if you were to extract it out into shared methods they'd be like one line. I think the current way of having `corners` get "filled in" with the projection extents results in the cleanest overall code.